### ISSUE TITLE
Materialize.js Tabs is not compatible with Angular Router using "Hash Strategy"

### ISSUE THREAD

**Issue by IAMtheIAM** [Dogfalo/materialize#5297]
Materialize.js Tabs is not compatible with Angular Router using "Hash Strategy", or any links prepended by `#`

## Expected Behavior

Materialize.js Tabs should be compatible with links prepended with "#"

## Current Behavior

[This line](https://github.com/Dogfalo/materialize/blob/17046f4fecca8308bfd2ca542d81814ca139bae8/js/tabs.js#L86) throws a js error when it encounters a link using "Hash Strategy" where the href is prepended by `#`

For example:

```html
<ul class="tabs">
      <li class="tab"><a ng-reflect-router-link="../packages" href="#/billing/packages">Membership Catalog</a></li>
      <li class="tab active"><a class="active" ng-reflect-router-link="../checkout" href="#/billing/checkout">Secure Checkout</a></li>
   </ul>
```

This throws the error:

```js
zone.js:196 Uncaught Error: Syntax error, unrecognized expression: #/billing/checkout
    at Function.Sizzle.error (sizzle.js:1042)
    at Sizzle.tokenize (sizzle.js:1694)
    at Sizzle.select (sizzle.js:2121)
    at Function.Sizzle [as find] (sizzle.js:346)
    at jQuery.fn.init.find (findFilter.js:81)
    at new jQuery.fn.init (init.js:93)
    at jQuery (core.js:34)
    at HTMLUListElement.eval (materialize.js:2658) <------- this is what causes it
    at Function.each (core.js:298)
    at jQuery.fn.init.each (core.js:93)
```

## Possible Solution

Remove the logic relating to preselecting tabs when a `#` is prepended to the `href`. Change preselect tab strategy to only use `class="active"`.

It seems like unnecessary code to allow users to either add class `active` manually, or to add a `#` in the `href` and have the jquery add `active` to that very same element. Users don't need two ways to do the same thing in the case. Adding `active` to the class in the HTML is more clear as to what is going to happen and will eliminate any collisions with other JS implementations that rely on `#` in the `href`

## Steps to Reproduce (for bugs)

1. Create page with materialize tabs 
2. Make a tab have an href starting with `#` i.e. `href="#mylink"`
3. Load page 

## Context

Trying to use the tabs to navigate between components/pages

**Comment by inigojpuente** [comment]
You should initialize the jQuery plugin inside the component as per [materialize.css documentation](http://materializecss.com/tabs.html)
In the .ts file of the component where you have the tab functionality implemented:
1. Add the following declaration at the top of the file: `declare var $:any;`
2. Inside the ngOnInit() add: `$(document).ready(function(){
    $('ul.tabs').tabs();
  });`

**Comment by Dogfalo** [comment]
This was caused by the slashes i believe, not the `#`. Fixed by #5292
